jdbc/src/main/java/org/springframework/jdbc/core/JdbcTemplate.java

```diff
@@ -20,60 +23,75 @@ public JdbcTemplate(final DataSource dataSource) {
         this.dataSource = dataSource;
     }
 
+    public <T> T queryForObject(final String sql, final RowMapper<T> rowMapper, final Object... args) {
+        final List<T> results = query(sql, rowMapper, args);
+
+        if (results.size() > 1) {
+            throw new IncorrectResultSizeDataAccessException(1, results.size());
+        }
+        if (results.isEmpty()) {
+            throw new EmptyDataAccessException();
+        }
+
+        return results.get(0);
+    }
+
     public <T> List<T> query(final String sql, final RowMapper<T> rowMapper, final Object... args) {
-        try (final Connection connection = dataSource.getConnection();
+        return executeInternal(sql, pstmt -> queryMultipleResults(pstmt, rowMapper), args);
+    }
```

Comment:
함수형 인터페이스를 람다식으로 사용해보는 건 어떨까요? 
```suggestion
        return executeInternal(sql, pstmt -> {
             final ResultSet resultSet = pstmt.executeQuery();
             final List<T> results = new ArrayList<>();
             while (resultSet.next()) {
                  results.add(rowMapper.mapRow(resultSet));
              }
             return results;
             };
    }
```
그럼 try-catch문으로 감쌀 필요가 없을 것 같아요.

Reply:
헉 확실히 좋아보여서 반영을 한 번 해보았으나...! 
저는 뎁스가 두 번 들어가는 상황 (람다식 시작 bracket 안에 while로 뎁스가 두 번 들어가는 것) 이 조금 신경쓰이네요... 😢 
객체지향 생활체조에 절여진 뇌 탓일까요..?

그래서 람다식을 활용하는 부분들 중에서 뎁스가 두 번 들어가지 않는 것들은 모두 메서드 분리한 것을 없앴고, 이 부분만 메서드를 호출하는 방식을 그대로 두었습니다!

그래도 확실히 이레의 리뷰 덕분에 `executeInternal` 부분에서만 try-catch가 필요해지고, 다른 부분들은 더 깔끔하게 try-catch 없이 코드를 작성할 수 있었어요 👍 👍 (❤ ω ❤)

Reply:
```
return executeInternal(sql, pstmt -> {
	             final ResultSet resultSet = pstmt.executeQuery();
	             final List<T> results = new ArrayList<>();
	             while (resultSet.next()) {
	                  results.add(rowMapper.mapRow(resultSet));
	              }
	             return results;
	             };
	    }
```
이 부분에서 뎁스가 두번 들어가는 부분은 없지 않나요? 브라켓 안에 while문은 뎁스라고 생각을 못했네용.
그런데 이것은 개인의 견해 차이라고 생각합니다. 저는 개인적으로 나누지 않는 것이 가독성이 더 좋다고 생각헸었어요
하지만 바론의 입장에서 메서드를 분리하는 것이 더 명확하다고 생각한다면(네이밍을 줌으로써 확실히 어떤 코드인지 보여줄 수 있는 것도 맞는 것 같아요) 그렇게 진행하도록 해주세용!

Reply:
엇 저도 람다식 선언부에서 시작하는 중괄호를 뎁스라고 생각할 수 있을지는 더 고민해 볼 필요가 있겠네용 🤔 
사실 하나의 새로운 메서드를 선언하는 것이니, 이중 뎁스라고 보기 어려울 것 같기도 해요! 

이레의 답변도 이해가 되고, 저 메서드가 `query` 내에서만 사용되니 람다식으로 두어도 좋을 것 같습니다 👍